### PR TITLE
add(java): abstract class and interface

### PR DESCRIPTION
**Pull request by goroyal**

### CODE REVIEW

**Comment by Kimi-Gao** [comment]
- add `docs: java` label in `.github/labler.yml` for github actions
- add path configuration in `sidebars.json` and `docusaurus.config.js`

**Comment by goroyal** [comment]
OK

**Comment by goroyal** [comment]
> OK



> * add `docs: java` label in `.github/labler.yml` for github actions
> * add path configuration in `sidebars.json` and `docusaurus.config.js`

OK

File: docs/java/1.basic/abstract-class-interface.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+(3)继承了抽象类的子类只能看到抽象类中`public`和`protected`类型的变量和方法，对于非`abstract`的方法不需要重写。而接口中定义的所有方法必须重写，定义的常量会影响所有实现它的类，所以，一般不推荐在接口里定义常量，且接口规模应尽可能小。如果需要抽象出多个方法，不应该把这些方法都放在接口里，而是采用多个接口+接口继承的形式。这点在Effective Java里面也有提到。
+
+(4)接口是对行为的一种抽象，而抽象类是对类的抽象，包括属性、方法。继承抽象类的类往往是具有一些相似特点的类，而实现接口的类可以跨不同的域，仅仅实现了接口定义的契约。类继承抽象类像是一个”is-a”特点，类实现接口像是”like-a”特点。
+
+(5)在设计时，对接口往往是自上而下的，先定义接口行为，然后再针对其做具体实现；抽象类往往是自下而上的，我们先知道子类后才对其进行抽象出父类。
+
+(6)我们常常看到Java程序里有定义的一些空接口，那么空接口是什么作用呢？
+
+空接口的主要是用来做判断的，也就是作为一个标记。为了判断某一个类是否满足其筛选条件时可以做一个空接口，然后利用instanceof方法来判断某一类是否使用了该接口，以达到你要筛选指定类型类的需求。
```
Comment (Kimi-Gao) [inline docs/java/1.basic/abstract-class-interface.md]: The above content is a bit `硬核`, introducing the abstract class and interface at the beginning will be more the summary more understandable for the beginner. Adding some images will be nice.

File: docs/java/1.basic/abstract-class-interface.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## 参考资料
+Effective Java
```
Comment (Kimi-Gao) [inline docs/java/1.basic/abstract-class-interface.md]: Add author and link

Comment (goroyal) [reply docs/java/1.basic/abstract-class-interface.md]: How?

File: docs/java/1.basic/abstract-class-interface.md (hunk 1 above)
Comment (goroyal) [reply docs/java/1.basic/abstract-class-interface.md]: 以后会加一些初级的内容

File: docs/java/1.basic/abstract-class-interface.md (hunk 2 above)
Comment (Kimi-Gao) [reply docs/java/1.basic/abstract-class-interface.md]: like this in md:
`1. [Effective Java, By Joshua Bloch](https://book.douban.com/subject/30412517/)`

**Pull request merged**